### DISCUSSION TITLE
TLS support from the box

### DISCUSSION THREAD

**Discussion by silvestrpredko** [tokio-rs/axum#1987] [Ideas]
I have seen examples of TLS support, but I couldn't understand why it's not added to the Axum.

Sometimes it is convenient to just pass a certificate and get working HTTPS from the box.

By the way `axum_server` does his job. But it's not convenient. For example, axum_server doesn`t have a with_graceful_shutdown after the serve was called.

I would like to add TLS support.
Or if it doesn't make sense could you please explain it to me?

**Comment by davidpdrsn** [comment]
TSL support won't be built into axum.

axum intentionally doesn't have a server built-in (`axum::Server` is a re-export of `hyper::Server`) and instead relies on the ecosystem for that.
